cocos/particle/line.ts

```diff
@@ -22,62 +22,59 @@
  THE SOFTWARE.
 */
 
-import { ccclass, help, executeInEditMode, menu, tooltip, displayOrder, type, serializable, range } from 'cc.decorator';
-import { Material, Texture2D } from '../asset/assets';
-import { Component } from '../scene-graph';
-import { Vec3, Vec2, Vec4, cclegacy } from '../core';
+import { ccclass, help, executeInEditMode, menu, tooltip, displayOrder, type, serializable, range, visible, override, displayName } from 'cc.decorator';
+import { Material } from '../asset/assets';
+import { Vec3, cclegacy, Vec4, Vec2 } from '../core';
 import { LineModel } from './models/line-model';
 import { builtinResMgr } from '../asset/asset-manager';
 import CurveRange from './animator/curve-range';
 import GradientRange from './animator/gradient-range';
-import { IMaterialInstanceInfo, MaterialInstance } from '../render-scene/core/material-instance';
-
-const _matInsInfo: IMaterialInstanceInfo = {
-    parent: null!,
-    owner: null!,
-    subModelIdx: 0,
-};
+import { ModelRenderer } from '../misc';
 
 const CC_USE_WORLD_SPACE = 'CC_USE_WORLD_SPACE';
-const define = { CC_USE_WORLD_SPACE: false };
+const CC_USE_WORLD_SCALE = 'CC_USE_WORLD_SCALE';
+const define = { CC_USE_WORLD_SPACE: false, CC_USE_WORLD_SCALE: true };
 
 @ccclass('cc.Line')
 @help('i18n:cc.Line')
 @menu('Effects/Line')
 @executeInEditMode
-export class Line extends Component {
-    @type(Texture2D)
-    private _texture = null;
+export class Line extends ModelRenderer {
+    @type(Material)
+    @displayOrder(1)
+    @tooltip('i18n:line.material')
+    @displayName('Material')
+    get lineMaterial () {
+        return this.getMaterial(0);
+    }
 
-    /**
-     * @zh 显示的纹理。
-     */
-    @type(Texture2D)
-    @displayOrder(0)
-    @tooltip('i18n:line.texture')
-    get texture () {
-        return this._texture;
+    set lineMaterial (val) {
+        this.setMaterial(val, 0);
     }
 
-    set texture (val) {
-        this._texture = val;
-        if (this._materialInstance) {
-            this._materialInstance.setProperty('mainTexture', val);
+    protected _onMaterialModified (index: number, material: Material | null) {
+        super._onMaterialModified(index, material);
+        const matIns = this.getMaterialInstance(0);
+        if (matIns) {
+            define[CC_USE_WORLD_SPACE] = this.worldSpace;
+            matIns.recompileShaders(define);
+            if (this._models[0]) {
+                const lineModel = this._models[0] as LineModel;
+                lineModel.updateMaterial(matIns);
+            }
         }
     }
-    @serializable
-    private _material: Material | null = null;
-    private _materialInstance: MaterialInstance | null = null;
 
+    @override
+    @visible(false)
     @type(Material)
```

Comment:
seems `@type(Material)` not correct. It's a array. Should it be `@type([Material])`?

Reply:
@type(Material) is in parent class

Reply:
removed type